types/ramda/index.d.ts

```diff
@@ -278,18 +278,18 @@ export function append<T>(el: T): <T>(list: readonly T[]) => T[];
  * Applies function fn to the argument list args. This is useful for creating a fixed-arity function from
  * a variadic function. fn should be a bound function if context is significant.
  */
-export function apply<F extends (...args: readonly any[]) => any>(fn: F, args: Parameters<F>): ReturnType<F>;
-export function apply<F extends (...args: readonly any[]) => any>(fn: F): (args: Parameters<F>) => ReturnType<F>;
+export function apply<F extends (...args: any[]) => unknown>(fn: F, args: Parameters<F>): ReturnType<F>;
```

Comment:
Great work on this cleanup! I was wondering if we shouldn't extract `(...args: any[]) => unknown` to some type like `AnyFunction` to be a bit more expressive?

Reply:
Yeah, maybe... I had actually deprecated a bunch of helper types in the overhaul PR, some because they were very simple...

In this case at least I do think it makes sense to add a helper type, so people don't get it wrong in the future.

That said, I'd probably prefer to make it take arguments... something like:
```typescript
type FnSomething<Args extends unknown[] = any[], Return extends unknown = unknown> =
  (...args: Args) => Return;
```
Not sure what it should be named though

Reply:
Worth noting there *is* already something similar (`Fn`) - IIRC it was added quite recently, but for a specific purpose. It could be repurposed though.

Reply:
Also worth noting there are some places which return a universal function - however in those places you'd want a different type (probably `(...args: unknown[]) => unknown`).
Not that it actually matters that much... I'm not sure how much it matters actually.
I think `...args: any[]` *might* affect return type inference in some places so *maybe* `...args: unknown[]` is better... but of course it doesn't hurt to return a type that should theoretically (?) be better

Reply:
In the cases where it really is just "Any Function" I'd actually use a really simple type and just name it that. The focus here should be readability, not reuse.